source/agora/consensus/EnrollmentPool.d

```diff
@@ -239,18 +239,51 @@ public class EnrollmentPool
 
     /***************************************************************************
 
-        Get the unregistered enrollments in the block
+        Get the height when the enrollment will be available
+
+        Params:
+            enroll_hash = key for the available block height
+
+        Returns:
+            the available block height, or 0 if the enrollment isn't found
+
+    ***************************************************************************/
+
+    public Height getAvailableHeight (const ref Hash enroll_hash)
+        @trusted nothrow
+    {
+        try
+        {
+            auto results = this.db.execute("SELECT avail_height " ~
+                "FROM enrollment_pool WHERE key = ?", enroll_hash.toString());
+            if (results.empty)
+                return Height(0);
+
+            return Height(results.oneValue!(size_t));
```

Comment:
When it has multiple records, it will return a record with the smallest 'avail_height'?
However, 'avail_height' is not a primary key, so it should be checked.

Reply:
> When it has multiple records, it will return a record with the smallest 'avail_height'?
> However, 'avail_height' is not a primary key, so it should be checked.

The `enroll_hash` which is a UTXO hash is a primary key for the table, so the `avail_height` for the key is only one. Thanks.

Reply:
Oh, I see. We can always register only one data in advance.